aips/aip-116.md

```diff
@@ -242,6 +234,40 @@ sha3_256(b"SIGN_IN_WITH_APTOS::") || <siwa_message>
 
 where `<siwa_message>` is the plain text construction of a SIWA message explained in the [Message Construction](#message-construction) section.
 
+### Signature and Message Verification
+
+The verification process consists of two equally important components:
+
+1. **Signature verification** - Cryptographically ensures the authenticity of a signature against the `message`.
+2. **SIWA Message verification** - Ensures the content of the `AptosSignInOutput` message adheres to the expected fields.
+
+#### Signature Verification
+
+Signature verification must consist of three parts:
+
+- `publicKey` - The public key of the user that is signing in.
+- `signature` - The signature to verify.
+- `message` - The SIWA message to verify.
+
+To verify the `signature`, the `publicKey` should be used to cryptographically verify the `signature` against the `message`. This verification should be stateless.
+
+#### Message Verification
+
+The message verification process must consist of the following parts:
+
+- `publicKey` - The public key of the user that is signing in. Should be the `publicKey` used in the `signature` verification step above.
+- `expected` - The `AptosSignInInput` fields that are expected to be present in the `message`.
+- `message` - The SIWA message to verify.
+
+The verification process must perform the following steps:
+
+1. Parse the `message` into an `AptosSignInInput` object.
+2. Compare the `expected` fields against the parsed `AptosSignInInput` object. If fields are present in the `expected` object, they must match the corresponding fields in the parsed `AptosSignInInput` object.
+3. Ensure that the `publicKey`'s authentication key's originating address is equal to the `address` field in the parsed `AptosSignInInput` object.
+4. If there is no originating address, ensure that the `publicKey`'s authentication key is equal to the `address` field in the parsed `AptosSignInInput` object.
+
+Implementation examples for both signature and message verification can be found in the [Verification Reference Implementations](#verification-reference-implementations) section.
+
 ### Modifications to the Wallet Standard
 
 This AIP also extends the [Wallet Standard](https://github.com/aptos-labs/wallet-standard), which is a chain-agnostic set of interfaces and conventions that aim to improve how applications interact with injected wallets.
```

Comment:
Isn't this the **Aptos** wallet standard that is being extended? Which is **not** actually chain agnostic...